src/cryptsetup/cryptsetup.c

```diff
@@ -132,12 +132,16 @@ static int parse_one_option(const char *option) {
 
         } else if ((val = startswith(option, "keyfile-offset="))) {
 
-                r = safe_atou(val, &arg_keyfile_offset);
+                r = safe_atou64(val, &arg_keyfile_offset);
                 if (r < 0) {
                         log_error_errno(r, "Failed to parse %s, ignoring: %m", option);
                         return 0;
                 }
 
+                if ((size_t) arg_keyfile_offset != arg_keyfile_offset)
+                        /* https://gitlab.com/cryptsetup/cryptsetup/issues/359 */
+                        log_warning("keyfile-offset= value truncated to %zu", (size_t) arg_keyfile_offset);
```

Comment:
hmm, we never permitted the truncated version, did we? safe_atou() after all just refused to parse this, hence we failed entirely? I think we should just fail if the user specifies a value >32bit on a 32bit arch, instead of warn and truncate

Reply:
I meant that somebody might have created the key at the wrong offset using some other tool. But that's overthinking the problem. I'll adjust to refuse.